### PR TITLE
refactor: swap ID from u64 to `SizedAsciiString<64>`

### PR DESCRIPTION
**Pull request by ra0x3**
Thanks for opening a PR with the Fuel Indexer project, please review the **Checklist** below and ensure you've completed all of the necessary steps to make this PR review as painless as possible.

### Checklist
- [ ] Ensure your top-level commit message is inline with our [contributor guidelines](./CONTRIBUTING.md).
- [ ] Please add proper labels.
- [ ] If there is an issue associated with this PR, please link the issue (right-hand sidebar)
- [ ] If there is not an issue associated with this PR, add this PR to the "Fuel Indexer" project (right-hand sidebar)
- [ ] Please allow Codeowners at least 24 hours to do a first-pass review.
- [ ] Please add thorougly detailed testing steps below.
- [ ] Please keep your Changelog message short and sweet.

### Description
- I noticed quite a bit of weirdness whilst working on this - so we won't be cutting 0.20 immediately until we add some more (relevant, non-redundant, useful) testing 
- PR does a few things
  - Does not optimize usage of `.clone()` https://github.com/FuelLabs/fuel-indexer/issues/1278
  - Refactors the project to use `SizedAsciiString<64>` IDs instead of `u64` IDs
    -  With `SizedAsciiString<64>` aliased as `UID`
  - Removes simple-wasm indexer
    - It wasn't testing anything we weren't already testing
  - Also includes more work from https://github.com/FuelLabs/fuel-indexer/issues/1194
    - We don't use any other DB anymore so removed "postgres" suffix from test modules 
    - Moves the fuel-indexer-test indexer from fuel-indexer-tests/components/indices/ to fuel-indexer-tests/indexers
      - simple-wasm too
    - Removes redundant tests (database) that have not detected much breakage in the past
    - Removes the graphql_schema test module
      - Our GraphQL layer is gonna get an entire makeover per https://github.com/FuelLabs/fuel-indexer/issues/1145
        - Grantees are using the SQL endpoint
   - Updates our trybuild tests (which have not been running on CI :/ )
     - Re-enables these on CI
   - Moves `IndexMetadataEntity` to `fuel_indexer_types`
     - And makes `fuel_indexer_lib` dependent on `fuel_indexer_types`
       - `fuel_indexer_types` is now the base crate (previously was `fuel_indexer_lib`)

### Testing steps

- As this was a very low-level/granular change, CI for everything should pass
- Start your indexer service on beta-4 `cargo run --bin fuel-indexer -- run --run-migrations --fuel-node-host beta-4.fuel.network --fuel-node-port 80`
- Deploy the default indexer `forc index new indexer-test --namespace fuellabs && cd indexer-test && forc index deploy`
- Let this index to around block 250K, you should have no warnings for constraint errors

### Changelog

- refactor: swap ID from u64 to `SizedAsciiString<64>`

### CODE REVIEW

File: packages/fuel-indexer/src/executor.rs [hunk 1]
```diff
@@ -687,7 +688,13 @@ impl WasmIndexExecutor {
             let schema_version_from_wasm = ffi::get_version(&mut store_mut, &instance)?;
 
             if schema_version_from_wasm != schema_version {
-                return Err(IndexerError::Unknown(format!("Schema version from WASM {schema_version_from_wasm} does not match schema version {schema_version}")));
+                // Just an FYI this error below is not getting unwrapped into a panic...
```
Comment (ra0x3) [inline packages/fuel-indexer/src/executor.rs:691]: FYI @lostman

File: packages/fuel-indexer-types/src/lib.rs [hunk 2]
```diff
@@ -21,102 +24,20 @@ pub trait TypeId {
...
     pub use crate::receipt::*;
     pub use crate::scalar::*;
-    pub use crate::{TypeId, FUEL_TYPES_NAMESPACE};
+    pub use crate::{
+        type_id, Bech32Address, Bech32ContractId, Bits256, Identity, SizedAsciiString,
+        TypeId, FUEL_TYPES_NAMESPACE,
+    };
 }
 
-#[cfg(test)]
```
Comment (ra0x3) [inline packages/fuel-indexer-types/src/lib.rs:29]: Not even sure we ever used these conversions

File: packages/fuel-indexer-tests/tests/graphql_server.rs [hunk 3]
```diff
@@ -256,24 +259,6 @@ async fn test_filtering() {
 
     mock_request("/ping").await;
 
-    // ID selection
```
Comment (ra0x3) [inline packages/fuel-indexer-tests/tests/graphql_server.rs:259]: - This part of the test was giving problems
- Something tells me it's related to passing the `ID` in here manually where `ID` was simply 1 - however, now, ID is now calculated differently . 
- The fact that everything else works kinda lets me know its this specific test

Comment (deekerno) [reply packages/fuel-indexer-tests/tests/graphql_server.rs:259]: Can we create an issue to ensure that we re-add an ID selection test at some point?

File: packages/fuel-indexer-plugin/src/wasm.rs [hunk 4]
```diff
@@ -75,15 +75,15 @@ pub trait Entity<'a>: Sized + PartialEq + Eq + std::fmt::Debug {
         }
     }
 
-    fn load(id: u64) -> Option<Self> {
+    fn load(id: UID) -> Option<Self> {
         unsafe {
-            let buf = id.to_le_bytes();
-            let mut buflen = (buf.len() as u32).to_le_bytes();
+            let buff = bincode::serialize(&id.to_string()).unwrap();
```
Comment (deekerno) [inline packages/fuel-indexer-plugin/src/wasm.rs:80]: I _think_ we can remove this `.to_string()` call here as `SizedAsciiString` provides an `impl` for `Serialize`: https://github.com/FuelLabs/fuels-rs/blob/df21541b679bb90a1ee3e54f34a67b7037b40be7/packages/fuels-core/src/types/core/sized_ascii_string.rs#L162-L169.

Additionally, can we replace the `.unwrap()` with an `.expect()`?

File: packages/fuel-indexer/src/ffi.rs [hunk 5]
```diff
@@ -258,22 +284,30 @@ impl<'a> WasmArg<'a> {
         })
     }
 
+    pub fn store(&mut self) -> &mut Store {
```
Comment (deekerno) [inline packages/fuel-indexer/src/ffi.rs:288]: Could we add a doc comment for this new function?

**Review by deekerno — CHANGES_REQUESTED** [review]
Running through your testing steps now; left a few questions/suggestions in the meantime.

File: packages/fuel-indexer-api-server/src/uses.rs [hunk 6]
```diff
@@ -316,11 +316,16 @@ pub(crate) async fn register_indexer_assets(
...
+        if let Err(e) = tx
+            .send(ServiceRequest::Reload(ReloadRequest {
+                namespace,
+                identifier,
+            }))
+            .await
+        {
+            error!("Failed to send ServiceRequest::Reload: {e:?}");
+            return Err(e.into());
+        }
```
Comment (lostman) [inline packages/fuel-indexer-api-server/src/uses.rs:328]: I wonder whether there is a better way of handling this pattern.

The error returned by `?` is generally logged somewhere up. We have a couple of error messages that repeat this way.

The advantage of the `error!` macro is the source location attribution.

Perhaps there is some way to have both concise code and precise location.

File: packages/fuel-indexer-tests/indexers/fuel-indexer-test/fuel_indexer_test.yaml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 namespace: fuel_indexer_test
 fuel_client: ~
-graphql_schema: packages/fuel-indexer-tests/components/indices/fuel-indexer-test/schema/fuel_indexer_test.graphql
+graphql_schema: packages/fuel-indexer-tests/indexers/fuel-indexer-test/schema/fuel_indexer_test.graphql
```
Comment (lostman) [inline packages/fuel-indexer-tests/indexers/fuel-indexer-test/fuel_indexer_test.yaml:3]: `indexers` 👍

**Review by deekerno — APPROVED** [review]

File: packages/fuel-indexer-api-server/src/uses.rs (hunk 6 above)
Comment (ra0x3) [reply packages/fuel-indexer-api-server/src/uses.rs:328]: @lostman 
- I agree it seems a bit...repetitive? 
- I think the main reason I bubbled things up in this way (and in executor/service) was to be almost annoyingly explicit with the path of the error (where we can trace the error's path just via looking at the logs)
  - I'm open to better ways to accomplish this

**Pull request merged**